server/cache/cache.go

```diff
@@ -24,6 +24,11 @@ type Cache struct {
 	loginAttemptsExpiration         time.Duration
 }
 
+type ResourceCachedState struct {
+	appSyncStatus  *appv1.SyncStatus
+	resourceStatus *appv1.ResourceStatus
```

Comment:
Should not be pointer type

Reply:
I am pretty sure this needs to be capital letter (external) properties. Because they are marshalled and unmarshalled later on.